libafl/src/observers/value.rs

```diff
@@ -88,3 +96,15 @@ where
         &self.name
     }
 }
+
+#[cfg(feature = "std")]
+impl<'a, T: Hash> ObserverWithHashField for ValueObserver<'a, T>
+where
+    T: Debug + Serialize + serde::de::DeserializeOwned,
+{
+    fn hash(&self) -> Option<u64> {
+        let mut s = DefaultHasher::new();
+        Hash::hash(self.value.as_ref(), &mut s);
```

Comment:
If you use AHasher here, it can be no_std.
Also we already use it in other places.
You can also use xxh3, another hash we already use in the lib (and which is a tiny bit faster)